eclipse-theia/theia
# resolve problemMatchers for tasks executed by vscode.tasks.executeTask api call

#### What it does
Fixes #9205 

Tasks executed with the direct call to runTask method (as it is in vscode.tasks.executeTask implementation) have no RunOptions, which also means they lose any ProblemMatchers that might be associated with this task.

Here we recreate RunOptions, resolving ProblemMatchers if they exist in TaskConfiguration.

#### How to test

1. Test on a vscode plugin with a task contribution  and problem matchers specified
2. Task should produce output matching with problemMatcher regexp
3. Run task with an vscode.tasks.executeTask call
4. Task should complete, "Problems" view should show 1 new record

#### Review checklist

- [ x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

Signed-off-by: Vladimir Fedotov <vfl@ispras.ru>
